### ISSUE TITLE
external video link crash

### ISSUE THREAD

**Issue by 1sad** [K1rakishou/Kuroba-Experimental#241]
release 0.3.0
oppening link to external video (catbox) crashes the app

video has no thumbnail so have to swipe to it from previous image, when on it n the viewer has "linked" thumbnail and play button, pressing play  starts loading gfx (no video, no thumbnail, just loading circle on black screen)

and app crashes

log (trying to send from the app works rarely for me) 
#239 is me as well I think, same thing happened on 0.2.12

```
=== LOGS(2020-10-31 13:16:55 UTC) ===
E/KurobaEx | ViewUtils: setHandlesColors() failure, error=Attempt to invoke virtual method 'java.lang.Class java.lang.Object.getClass()' on a null object reference
E/KurobaEx | ViewUtils: setHandlesColors() failure, error=Attempt to invoke virtual method 'java.lang.Class java.lang.Object.getClass()' on a null object reference
E/KurobaEx | ViewUtils: setHandlesColors() failure, error=Attempt to invoke virtual method 'java.lang.Class java.lang.Object.getClass()' on a null object reference
E/KurobaEx | ViewUtils: setHandlesColors() failure, error=Attempt to invoke virtual method 'java.lang.Class java.lang.Object.getClass()' on a null object reference
E/KurobaEx | ViewUtils: setHandlesColors() failure, error=Attempt to invoke virtual method 'java.lang.Class java.lang.Object.getClass()' on a null object reference
E/KurobaEx | AndroidUtils: view.getViewTreeObserver() is another viewtreeobserver! replacing with the new one
E/KurobaEx | AndroidUtils: view.getViewTreeObserver() is another viewtreeobserver! replacing with the new one
E/KurobaEx | ViewUtils: setHandlesColors() failure, error=Attempt to invoke virtual method 'java.lang.Class java.lang.Object.getClass()' on a null object reference
E/KurobaEx | ViewUtils: setHandlesColors() failure, error=Attempt to invoke virtual method 'java.lang.Class java.lang.Object.getClass()' on a null object reference
E/KurobaEx | ViewUtils: setHandlesColors() failure, error=Attempt to invoke virtual method 'java.lang.Class java.lang.Object.getClass()' on a null object reference
E/KurobaEx | ViewUtils: setHandlesColors() failure, error=Attempt to invoke virtual method 'java.lang.Class java.lang.Object.getClass()' on a null object reference
E/KurobaEx | AndroidUtils: view.getViewTreeObserver() is another viewtreeobserver! replacing with the new one
E/KurobaEx | ViewUtils: setHandlesColors() failure, error=Attempt to invoke virtual method 'java.lang.Class java.lang.Object.getClass()' on a null object reference
E/KurobaEx | ViewUtils: setHandlesColors() failure, error=Attempt to invoke virtual method 'java.lang.Class java.lang.Object.getClass()' on a null object reference
I/KurobaEx | WebmStreamingDataSource: detectLength(): 15189355
I/KurobaEx | WebmStreamingDataSource: opening, position: 0 length: -1
I/KurobaEx | WebmStreamingDataSource: close
E/KurobaEx | Chan: Unhandled exception
E/KurobaEx | Chan: java.lang.NoClassDefFoundError: com.google.android.exoplayer2.extractor.mp4.-$$Lambda$Mp4Extractor$IP-_bKEryH3151zsYMHu9R1OlAQ
E/KurobaEx | Chan: 	at com.google.android.exoplayer2.extractor.mp4.Mp4Extractor.processAtomEnded(Mp4Extractor.java:115)
E/KurobaEx | Chan: 	at com.google.android.exoplayer2.extractor.mp4.Mp4Extractor.read(Mp4Extractor.java:79)
E/KurobaEx | Chan: 	at com.google.android.exoplayer2.source.ProgressiveMediaPeriod$ExtractingLoadable.load(ProgressiveMediaPeriod.java:44)
E/KurobaEx | Chan: 	at com.google.android.exoplayer2.upstream.Loader$LoadTask.run(Loader.java:6)
E/KurobaEx | Chan: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
E/KurobaEx | Chan: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
E/KurobaEx | Chan: 	at java.lang.Thread.run(Thread.java:818)
E/KurobaEx | Chan: Unhandled exception
E/KurobaEx | Chan: java.lang.NoClassDefFoundError: com.google.android.exoplayer2.extractor.mp4.-$$Lambda$Mp4Extractor$IP-_bKEryH3151zsYMHu9R1OlAQ
E/KurobaEx | Chan: 	at com.google.android.exoplayer2.extractor.mp4.Mp4Extractor.processAtomEnded(Mp4Extractor.java:115)
E/KurobaEx | Chan: 	at com.google.android.exoplayer2.extractor.mp4.Mp4Extractor.read(Mp4Extractor.java:79)
E/KurobaEx | Chan: 	at com.google.android.exoplayer2.source.ProgressiveMediaPeriod$ExtractingLoadable.load(ProgressiveMediaPeriod.java:44)
E/KurobaEx | Chan: 	at com.google.android.exoplayer2.upstream.Loader$LoadTask.run(Loader.java:6)
E/KurobaEx | Chan: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
E/KurobaEx | Chan: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
E/KurobaEx | Chan: 	at java.lang.Thread.run(Thread.java:818)
E/KurobaEx | UNCAUGHT: Called from unhandled exception handler.
E/KurobaEx | UNCAUGHT: java.lang.NoClassDefFoundError: com.google.android.exoplayer2.extractor.mp4.-$$Lambda$Mp4Extractor$IP-_bKEryH3151zsYMHu9R1OlAQ
E/KurobaEx | UNCAUGHT: 	at com.google.android.exoplayer2.extractor.mp4.Mp4Extractor.processAtomEnded(Mp4Extractor.java:115)
E/KurobaEx | UNCAUGHT: 	at com.google.android.exoplayer2.extractor.mp4.Mp4Extractor.read(Mp4Extractor.java:79)
E/KurobaEx | UNCAUGHT: 	at com.google.android.exoplayer2.source.ProgressiveMediaPeriod$ExtractingLoadable.load(ProgressiveMediaPeriod.java:44)
E/KurobaEx | UNCAUGHT: 	at com.google.android.exoplayer2.upstream.Loader$LoadTask.run(Loader.java:6)
E/KurobaEx | UNCAUGHT: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
E/KurobaEx | UNCAUGHT: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
E/KurobaEx | UNCAUGHT: 	at java.lang.Thread.run(Thread.java:818)
E/KurobaEx | UNCAUGHT: ------------------------------
E/KurobaEx | UNCAUGHT: END OF CURRENT RUNTIME MESSAGES
E/KurobaEx | UNCAUGHT: ------------------------------
E/KurobaEx | UNCAUGHT: Android API Level: 21
E/KurobaEx | UNCAUGHT: App Version: v0.3.0.4105
E/KurobaEx | UNCAUGHT: Called from unhandled exception handler.
E/KurobaEx | UNCAUGHT: java.lang.NoClassDefFoundError: com.google.android.exoplayer2.extractor.mp4.-$$Lambda$Mp4Extractor$IP-_bKEryH3151zsYMHu9R1OlAQ
E/KurobaEx | UNCAUGHT: 	at com.google.android.exoplayer2.extractor.mp4.Mp4Extractor.processAtomEnded(Mp4Extractor.java:115)
E/KurobaEx | UNCAUGHT: 	at com.google.android.exoplayer2.extractor.mp4.Mp4Extractor.read(Mp4Extractor.java:79)
E/KurobaEx | UNCAUGHT: 	at com.google.android.exoplayer2.source.ProgressiveMediaPeriod$ExtractingLoadable.load(ProgressiveMediaPeriod.java:44)
E/KurobaEx | UNCAUGHT: 	at com.google.android.exoplayer2.upstream.Loader$LoadTask.run(Loader.java:6)
E/KurobaEx | UNCAUGHT: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
E/KurobaEx | UNCAUGHT: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
E/KurobaEx | UNCAUGHT: 	at java.lang.Thread.run(Thread.java:818)
E/KurobaEx | UNCAUGHT: ------------------------------
E/KurobaEx | UNCAUGHT: END OF CURRENT RUNTIME MESSAGES
E/KurobaEx | UNCAUGHT: ------------------------------
E/KurobaEx | UNCAUGHT: Android API Level: 21
E/KurobaEx | UNCAUGHT: App Version: v0.3.0.4105
E/KurobaEx | UNCAUGHT: Development Build: Release
E/KurobaEx | UNCAUGHT: Phone Model: samsung SM-A310F
E/KurobaEx | UNCAUGHT: Development Build: Release
E/KurobaEx | UNCAUGHT: Phone Model: samsung SM-A310F


=== SETTINGS ===
Prefetching enabled: false
Hi-res thumbnails enabled: false
Youtube titles and durations parsing enabled: WIFI
Concurrent file loading chunks count: 2
WEBM streaming enabled: true
Saved files base dir info: Java API location: "/storage/emulated/0/Download/4chan", SAF location: "", active: Java API
Phone layout mode: PHONE
OkHttp IPv6 support enabled: false
OkHttp HTTP/2 support enabled: false
```

**Comment by K1rakishou** [comment]
Can you post a thread where this link appears?

**Comment by 1sad** [comment]
caution:
its probably likely most definitely porn

https://boards.4chan.org/v/thread/530422471#p530459038


also #239 has more about the file in the log i think, it doesnt even show the catbox url in this issues log

**Comment by K1rakishou** [comment]
Alright, I managed to fix the crash but apparently the video won't be able to be played on Android 5.

**Comment by 1sad** [comment]
that's fair, my phone is old, im glad app supports it at all

is it possible to implement a choice to have external linked videos go to external players from gallery? 
sort of like when clicking the link it asks you if you want to open it in a media player or a browser

**Comment by K1rakishou** [comment]
I guess something like that should be possible.

**Comment by K1rakishou** [comment]
I'll create a separate issue for that feature.
https://github.com/K1rakishou/Kuroba-Experimental/issues/242

**Comment by 1sad** [comment]
im not really demanding it or anything like that, just thinking out loud
linked videos are pretty rare, anyway
just thought id report the crashes, thank you

**Comment by K1rakishou** [comment]
Oh, okay then.

**Comment by K1rakishou** [comment]
Fixed in v0.3.1-beta and release.
